### PR TITLE
Make api more accessible, part 2

### LINKED ISSUE #262: Add straight path as method on `optimal_path()`
I am looking at the paths notebook, and I think it would be helpful if the straight path were added to the `optimal_path()` function to reduce some clutter.

My main goal is being able to remove the `straigh_path` function definition.

### LINKED ISSUE #263: Include new notebooks in documentation
- [ ] `notebooks/percolation.ipynb`
- [ ] `notebooks/multiple_paths.ipynb`

### PR DESCRIPTION
**Pull request by stefsmeets**
This PR adds some refinements to the paths api.

- Add repr to `Pathway`
- Add 'direct' option to `optimize_path()`
- Move plotter for paths directly to `plots.density`
- Add voxel/frac conversion methods on `Volume`

Closes #262
Closes #263 

### Todo

- [x] Add paths as argument on `Volume.plot()`, refactor `plots.plotly._paths` into `plots.plotly._density`
- [x] Add tests for direct path
- [x] Add tests for voxel/frac conversion, check 0.5 shift

Complemented by https://github.com/GEMDAT-repos/gemdat-notebooks/pull/8

### CODE REVIEW

File: src/gemdat/path.py [hunk 1]
```diff
@@ -30,12 +30,27 @@ class Pathway:
...
+            f'Steps: {len(self.sites)}',
+            f'Total energy: {self.total_energy:.3f} eV',
+        )
+
+        return '\n'.join(s)
+
+    @property
+    def total_energy(self):
+        """Return total energy for path."""
+        return sum(self.energy)
```
Comment (SCiarella) [inline src/gemdat/path.py:45]: Good suggestions!

File: src/gemdat/path.py [hunk 2]
```diff
@@ -203,7 +222,7 @@ def free_energy_graph(F: np.ndarray,
 
 
 _PATHFINDING_METHODS = Literal['dijkstra', 'bellman-ford', 'minmax-energy',
-                               'dijkstra-exp']
+                               'dijkstra-exp', 'direct']
```
Comment (SCiarella) [inline src/gemdat/path.py:225]: What if we call it 'simple' ?

Comment (SCiarella) [reply src/gemdat/path.py:225]:
```suggestion
                               'dijkstra-exp', 'simple']
```

File: src/gemdat/path.py [hunk 3]
```diff
@@ -283,6 +302,7 @@ def multiple_paths(
         Coordinates of the stopping point
     method : str
         Method used to calculate the shortest path. Options are:
+        - 'direct': Shortest, unweighted path
```
Comment (SCiarella) [inline src/gemdat/path.py:305]:
```suggestion
        - 'simple': Shortest, unweighted path
```

File: src/gemdat/path.py [hunk 4]
```diff
@@ -341,6 +361,7 @@ def optimal_path(
         Coordinates of the stoping point
     method : str
         Method used to calculate the shortest path. Options are:
+        - 'direct': Shortest, unweighted path
```
Comment (SCiarella) [inline src/gemdat/path.py:364]:
```suggestion
        - 'simple': Shortest, unweighted path
```

File: src/gemdat/path.py [hunk 5]
```diff
@@ -351,14 +372,21 @@ def optimal_path(
...
     """
-
-    optimal_path = nx.shortest_path(
-        F_graph,
-        source=start,
-        target=stop,
-        weight='weight_exp' if method == 'dijkstra-exp' else 'weight',
-        method='dijkstra' if method in ('dijkstra-exp',
-                                        'minmax-energy') else method)
+    if method == 'direct':
```
Comment (SCiarella) [inline src/gemdat/path.py:375]:
```suggestion
    if method == 'simple':
```

File: src/gemdat/plots/plotly/_density.py [hunk 6]
```diff
@@ -210,6 +278,9 @@ def density(volume: Volume,
         else:
             plot_structure(structure, fig=fig)
 
+    if paths:
```
Comment (SCiarella) [inline src/gemdat/plots/plotly/_density.py:281]: I like this option to add the paths directly to `density` plot :+1: 
We should update the notebooks to this standard.

File: tests/integration/path_test.py [hunk 7]
```diff
@@ -22,6 +22,7 @@ def test_fractional_coordinates(vasp_path_vol, vasp_full_path):
 
 TEST_DATA = (
     # start, stop, method, expected
+    ((10, 4, 13), (21, 3, 10), 'direct', 6.3023933200094655),
```
Comment (SCiarella) [inline tests/integration/path_test.py:25]:
```suggestion
    ((10, 4, 13), (21, 3, 10), 'simple', 6.3023933200094655),
```

**Review by SCiarella — APPROVED** [review]
Looks good :+1: 
I suggest replacing `direct` with `simple` for the unweighted Dijkstra path.
And let's not forget to update the example notebooks to use the `density` function to plot the paths.

File: src/gemdat/plots/plotly/_density.py (hunk 6 above)
Comment (stefsmeets) [reply src/gemdat/plots/plotly/_density.py:281]: Yep, already did it!

**Pull request merged**